daemon.go

```diff
@@ -386,45 +391,24 @@ func main() {
 		log.Fatal("Graceful termination is not supported on your platform.")
 	}
 
-	watcher, err := fsnotify.NewWatcher()
+	///////////////////////////////
+
+	watcher, err := NewWatcher(*flagPolling)
```

Comment:
Personally I would have found it better to pass the global variables like ignored directories, `flagRecursive`, etc. to the watcher and not rely on global variables in `watcher.go` since it would make the component testable in the future. Do you think you can do that?

Reply:
Absolutely I can :)